internal/venice-common/src/main/java/com/linkedin/venice/writer/WriterChunkingHelper.java

```diff
@@ -110,7 +111,7 @@ public static ChunkedPayloadAndManifest chunkPayloadAndSend(
         putPayload.putValue = EMPTY_BYTE_BUFFER;
         putPayload.replicationMetadataPayload = chunk;
       }
-      chunkedKeySuffix.chunkId.chunkIndex = chunkIndex;
+      chunkedKeySuffix.chunkId.chunkIndex = chunkIndex + chunkedKeySuffixStartingIndex;
```

Comment:
Can we have some unit test for this new logic?

Reply:
Yes it is actually covered in the VeniceWriterTest.testReplicationMetadataChunking() where it putLargeRecord and validates both value and RMD manifest. The test actually failed before I correct the expected keyWithSuffix for first RMD chunk.